float64 curLinVelX
float64 curLinVelY